libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -279,6 +284,12 @@ bool ContractLevelChecker::checkFunctionOverride(FunctionDefinition const& _func
 		success = false;
 	}
 
+	if (!_super.isVirtual() && !_super.annotation().contract->isInterface())
```

Comment:
We could even think about adding a ``isVirtualOrInterfaceFunction`` that checks both - that might make it less likely to miss the fact that interface functions are implicitly virtual despite ``isVirtual`` returning ``false``? But yeah: as you like, I'd be fine without that.

Reply:
Would you then get rid of `isVirtual`?

Reply:
`hasVirtualSemantics` - `isMarkedVirtual`

Reply:
Or better in line with coding stile: `virtualSemantics` - `markedVirtual`

Reply:
Or maybe this will be the new ``canBeOverridden`` :-)? But yeah, I think it'd be best to split into two, but I don't mind the exact naming - e.g. ``virtualSemantics`` and ``markedVirtual`` would work for me.

Reply:
Or ``overridable()`` for that matter - but in the end that's the semantics of being ``virtual``, isn't it?